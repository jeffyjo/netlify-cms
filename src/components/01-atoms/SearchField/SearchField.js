import React from 'react'
import propTypes from 'prop-types'

import './SearchField.scss'

const SearchField = ({componentClass, action, value}) => {

  return (
    <input 
      type="search" 
      onChange={action}
      className={"a-search " + componentClass ? componentClass : ''}
      value={value || ''}
    />
  )
}

SearchField.propTypes = {
  componentClass: propTypes.string,
  action: propTypes.func,
  value: propTypes.string
}

export default SearchField