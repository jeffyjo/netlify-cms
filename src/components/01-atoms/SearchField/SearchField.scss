.a-search {
  padding: .5em;
  border-radius: .25em;
  border: none;
}