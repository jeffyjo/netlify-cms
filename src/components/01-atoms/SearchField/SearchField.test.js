import React from 'react'
import SearchField from './SearchField'
import renderer from 'react-test-renderer'

describe('Atom | SearchField', () => {
  let component = null
  let tree = null
  let action = () => true

  beforeEach(() => {
    component = renderer.create(
      <SearchField action={action} componentClass="search-field-class" />
    )
    tree = component.toJSON()
  })
  
  test('SearchField should render', () => {
    expect(tree).toMatchSnapshot()
  })
  
  test('SearchField should have CSSclass', () => {
    expect(tree.props.className).toEqual('search-field-class')
  })

  test('SearchField should trigger action', () => {
    const componentAction = tree.props.onChange()
    expect(componentAction).toBeTruthy()
  })

  test('SearchField should change value', () => {
    expect(tree).toMatchSnapshot()
    tree.props.value = 'This is a test'
    expect(tree).toMatchSnapshot()
  })

})