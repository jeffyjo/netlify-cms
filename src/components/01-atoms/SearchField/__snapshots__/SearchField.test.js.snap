// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atom | SearchField SearchField should change value 1`] = `
<input
  className="search-field-class"
  onChange={[Function]}
  placeholder="Search..."
  type="search"
  value=""
/>
`;

exports[`Atom | SearchField SearchField should change value 2`] = `
<input
  className="search-field-class"
  onChange={[Function]}
  placeholder="Search..."
  type="search"
  value="This is a test"
/>
`;

exports[`Atom | SearchField SearchField should render 1`] = `
<input
  className="search-field-class"
  onChange={[Function]}
  placeholder="Search..."
  type="search"
  value=""
/>
`;
